compiler/kernel/op.go

```diff
@@ -329,6 +330,39 @@ func (b *Builder) compileLeaf(o dag.Op, parent zbuf.Puller) (zbuf.Puller, error)
 		return meta.NewDeleter(b.octx, parent, pool, filter, pruner, b.progress, b.deletes), nil
 	case *dag.Load:
 		return load.New(b.octx, b.source.Lake(), parent, v.Pool, v.Branch, v.Author, v.Message, v.Meta), nil
+	case *dag.UserOpCall:
+		// Construct the "this" record value.
+		elems := make([]expr.RecordElem, len(v.Params))
+		for i, p := range v.Params {
+			val, err := compileExpr(v.Exprs[i])
+			if err != nil {
+				return nil, err
+			}
+			switch p := p.(type) {
+			case *dag.VarParam:
+				elems[i] = expr.RecordElem{Name: p.Name, Field: val}
+			case *dag.SpreadParam:
+				elems[i] = expr.RecordElem{Spread: val}
+			}
+		}
+		this, err := expr.NewRecordExpr(b.octx.Zctx, elems)
+		if err != nil {
+			return nil, err
+		}
+		op := user.New(parent, this)
+		exits, err := b.compileSeq(v.Body, []zbuf.Puller{op})
+		if err != nil {
+			return nil, err
+		}
+		var exit zbuf.Puller
+		if len(exits) == 1 {
+			exit = exits[0]
+		} else {
+			// This can happen when output of the body
+			// is a fork or switch.
+			exit = combine.New(b.octx, exits)
+		}
+		return exit, nil
```

Comment:
Nit: Simplify.
```suggestion
		if len(exits) > 1 {
			// This can happen when output of the body
			// is a fork or switch.
			return combine.New(b.octx, exits), nil
		}
		return exits[0], nil
```